src/Planners/src/ContactList.cpp

```diff
@@ -0,0 +1,231 @@
+{
+    m_defaultName = defaultName;
+}
+
+const std::string &ContactList::defaultName() const
+{
+    return m_defaultName;
+}
+
+void ContactList::setDefaultContactType(const ContactType &type)
+{
+    m_defaultContactType = type;
+}
+
+const ContactType &ContactList::defaultContactType() const
+{
+    return m_defaultContactType;
+}
+
+bool ContactList::addContact(const Contact &newContact)
+{
+    if (newContact.activationTime > newContact.deactivationTime)
+    {
+        std::cerr << "[ContactList::addContact] The activation time cannot be greater than the deactivation time." <<std::endl;
+        return false;
+    }
+    using iterator = std::set<Contact, ContactCompare>::iterator;
+    std::pair<iterator,bool> res = m_contacts.insert(newContact);
+    if (!res.second)
+    {
+        std::cerr << "[ContactList::addContact] Failed to insert new element.";
+        if (res.first != end())
+        {
+            std::cerr << " The new contact (activationTime: " << newContact.activationTime << " ";
+            std::cerr << "deactivationTime: " << newContact.deactivationTime << ") is not compatible";
+            std::cerr << " with an element already present in the list (activationTime: " << res.first->activationTime << " ";
+            std::cerr << "deactivationTime: " << res.first->deactivationTime << ")" << std::endl;
+        }
+        return false;
+    }
+    return true;
+}
+
+bool ContactList::addContact(const iDynTree::Transform &newTransform, double activationTime, double deactivationTime)
+{
+    Contact newContact;
+    newContact.pose = newTransform;
+    newContact.activationTime = activationTime;
+    newContact.deactivationTime = deactivationTime;
+    newContact.name = m_defaultName;
+    newContact.type = m_defaultContactType;
+
+    return addContact(newContact);
+}
+
+ContactList::const_iterator ContactList::erase(const_iterator iterator)
+{
+    return m_contacts.erase(iterator);
+}
+
+ContactList::const_iterator ContactList::begin() const
+{
+    return m_contacts.begin();
+}
+
+ContactList::const_iterator ContactList::cbegin() const
+{
+    return m_contacts.cbegin();
+}
+
+ContactList::const_reverse_iterator ContactList::rbegin() const
+{
+    return m_contacts.rbegin();
+}
+
+ContactList::const_reverse_iterator ContactList::crbegin() const
+{
+    return m_contacts.crbegin();
+}
+
+ContactList::const_iterator ContactList::end() const
+{
+    return m_contacts.end();
+}
+
+ContactList::const_iterator ContactList::cend() const
+{
+    return m_contacts.cend();
+}
+
+ContactList::const_reverse_iterator ContactList::rend() const
+{
+    return m_contacts.rend();
+}
+
+ContactList::const_reverse_iterator ContactList::crend() const
+{
+    return m_contacts.crend();
+}
+
+const Contact &ContactList::operator[](size_t index) const
+{
+    assert(index < size());
+
+    if (index > (size() / 2 + (size() % 2 != 0))) //fancy way for doing std::ceil(size() / 2.0)
```

Comment:
Premature optimization :grin: I needed the ceil, but the division between integers gives the floor so I add one in case the remainder is not zero. The reason is that I wanted to avoid casting this division with a double. 
This ``if`` was meant to make this call slightly faster when used in a for loop. In fact, an ``std::set`` does not provide a random access operator. Hence, I used ``std::advance``, which basically increase by one the iterator for ``index`` times. I thought that if ``index`` is greater than half of the size, it is faster to reach the desired position by starting from the end. But this advantage would have been eaten up by the division itself.